react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    async function getItems() {
+      try {
+        setItems(await readItems());
+        setIsLoading(false);
+      } catch (err) {
+        setIsLoading(false);
+        setError(err);
+      }
```

Comment:
You can also use `finally` and put `setIsLoading(false);` in the `finally` block.